sources/packages/backend/apps/api/src/testHelpers/auth/bceid-helpers.ts

```diff
@@ -0,0 +1,47 @@
+import { TestingModule } from "@nestjs/testing";
+import { Institution, User } from "@sims/sims-db";
+import { getProviderInstanceForModule } from "@sims/test-utils";
+import { AppInstitutionsModule } from "../../app.institutions.module";
+import { BCeIDService } from "../../services";
+
+/**
+ * Mocks BCeID account info from BCeID service to return the passed BCeID account.
+ * @param testingModule nest testing module.
+ * @param user a persisted user object.
+ * @param displayName a persisted display name for user login.
+ * @param institution a persisted institution object.
+ * @param returnsNull indicates whether mocks a null response.
+ * @returns a persisted BCeID account object.
+ */
+export async function mockBCeIDAccountDetails(
+  testingModule: TestingModule,
+  user: User,
+  institution: Institution,
+  returnsNull?: boolean,
```

Comment:
What is the benefit of having an option to return null? Is it an attempt to "reset" the mock?

Reply:
Thanks for the explanation. Based on that we can have two methods.
The methods would be:
- mockBCeIDAccountUserFound
- mockBCeIDAccountUserNotFound
The above method should receive only the required parameters.